src/server/dfly_main.cc

```diff
@@ -96,6 +99,122 @@ namespace dfly {
 
 namespace {
 
+using util::http::TlsClient;
+
+const std::string_view kInvalidVersion{"unknown version"};
+
+std::string GetVersionString(const std::string& from) {
+  // The server sends a message such as {"latest": "0.12.0"}
+  const auto reg_match_expr = R"(\{\"latest"\:[ \t]*\"([0-9]+\.[0-9]+\.[0-9]+)\"\})";
+  VLOG(1) << "checking version '" << from << "'";
+  auto const regex = std::regex(reg_match_expr);
+  std::smatch match;
+  if (std::regex_match(from, match, regex) && match.size() > 1) {
+    // the second entry is the match to the group that holds the version string
+    return match[1].str();
+  } else {
+    return std::string{kInvalidVersion};
+  }
+}
+
+std::string GetRemoteVersion(ProactorBase* proactor, SSL_CTX* ssl_context, const std::string host,
+                             std::string_view service, const std::string& resource) {
+  namespace bh = boost::beast::http;
+  using ResponseType = bh::response<bh::string_body>;
+
+  TlsClient http_client{proactor};
+
+  http_client.set_connect_timeout_ms(2000);
+  auto version = proactor->Await([&] {
+    if (auto ec = http_client.Connect(host, service, ssl_context); !ec) {
```

Comment:
nit: this lambda is too long for a lambda :) 
or alternatively, maybe use 
```cpp
ec = .....
if (ec)
  return ec;
```
to keep the indentation in check.

Reply:
changed and this is no longer with a lambda in the function - moved out